### PR TITLE
feat: agent function

### PR DESCRIPTION
**Pull request by bettybas**

### CODE REVIEW

File: functions/agent/1.0/function.json [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+          "required": true
+        }
+      },
+      "name": "apiKey",
+      "label": "API Key",
+      "info": "The API key for the model"
+    },
+    {
+      "name": "prompt",
+      "label": "Prompt",
```
Comment (emiel-betty) [inline functions/agent/1.0/function.json:53]: Should we not make the prompt required?

File: functions/agent/1.0/index.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+      {}
+    );
+  
+    const { result } = await generativeAI({
+      authorization: {
+        apiKey,
+      },
+      endpoint: "text-to-text",
+
+      params: { prompt, variables: variableMap, model: { model: model, settings: { maxNewTokens: maxTokens, temperature: temperature / 100 } } },
```
Comment (emiel-betty) [inline functions/agent/1.0/index.js:16]: Can you do propper formatting

File: functions/agent/1.0/index.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    });
+  
+    console.log(result);
```
Comment (emiel-betty) [inline functions/agent/1.0/index.js:19]: Please remove the console log

File: functions/agent/1.0/function.json [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+        "type": "Text",
+        "validations": {
+          "required": true
+        }
+      },
+      "name": "apiKey",
+      "label": "API Key",
+      "info": "The API key for the model"
+    },
+    {
```
Comment (emiel-betty) [inline functions/agent/1.0/function.json:51]: Maybe add some info text here as well

File: functions/agent/1.0/function.json [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+      "name": "prompt",
+      "label": "Prompt",
+      "meta": {
+        "type": "MultilineText"
+      }
+    },
+    {
+      "name": "variables",
```
Comment (emiel-betty) [inline functions/agent/1.0/function.json:59]: Maybe add some info text here as well

File: functions/agent/1.0/function.json (hunk 1 above)
Comment (bettybas) [reply functions/agent/1.0/function.json:53]: True

**Review by emiel-betty — APPROVED** [review]

**Pull request merged**